### PR TITLE
Change `assert_layout_matches` function to match layouts sporadically.

### PR DESCRIPTION
**Pull request by vusirikala**

### CODE REVIEW

File: aptos-move/aptos-vm/src/move_vm_ext/respawned_session.rs [hunk 1]
```diff
@@ -129,12 +130,15 @@ pub fn assert_layout_matches(
...
     layout_2: Option<&MoveTypeLayout>,
 ) -> Result<(), PanicError> {
-    //TODO[agg_v2](optimize): Don't compare the layouts everytime. Do this operation sporadically
-    if layout_1 != layout_2 {
-        return Err(code_invariant_error(format!(
-            "Layouts don't match when they are expected to: {:?} and {:?}",
-            layout_1, layout_2
-        )));
+    if layout_1.is_some() || layout_2.is_some() {
+        let mut rng = rand::thread_rng();
```
Comment (georgemitenkov) [inline aptos-move/aptos-vm/src/move_vm_ext/respawned_session.rs:134]: Let's add a comment with a rationale why we do this sporadically?

**Review by georgemitenkov — APPROVED** [review]
Thanks! Let's also change the function name so it is clear what it means, e.g. `sporadically _assert_layout_matches` or `randomly_check_layout_matches` or smth like that. Then the comment can go away on line 128.

**Review by igor-aptos — APPROVED** [review]
can you rebase (there is a stack of 33 changes showing?), and address @georgemitenkov 's comments. otherwise looks good.

File: aptos-move/aptos-vm/src/move_vm_ext/respawned_session.rs [hunk 2]
```diff
@@ -129,12 +130,15 @@ pub fn assert_layout_matches(
     layout_1: Option<&MoveTypeLayout>,
     layout_2: Option<&MoveTypeLayout>,
 ) -> Result<(), PanicError> {
-    //TODO[agg_v2](optimize): Don't compare the layouts everytime. Do this operation sporadically
-    if layout_1 != layout_2 {
-        return Err(code_invariant_error(format!(
-            "Layouts don't match when they are expected to: {:?} and {:?}",
-            layout_1, layout_2
-        )));
+    if layout_1.is_some() || layout_2.is_some() {
```
[the remark this answers is not in the document]
Comment (vusirikala) [reply aptos-move/aptos-vm/src/move_vm_ext/respawned_session.rs:151]: Done.

File: aptos-move/aptos-vm/src/move_vm_ext/respawned_session.rs (hunk 1 above)
Comment (vusirikala) [reply aptos-move/aptos-vm/src/move_vm_ext/respawned_session.rs:152]: Done.

**Pull request merged**